lib/WTSI/NPG/HTS/PacBio/Sequel/RunPublisher.pm

```diff
@@ -185,10 +254,13 @@ sub publish_sequence_files {
 
   # Auxiliary files are kept for now but are not useful and so are
   # not marked as target.
-  my $is_aux = ($types eq $SEQUENCE_AUXILIARY) ? 1 : 0;
+  my $is_aux = ($type eq $SEQUENCE_AUXILIARY) ? 1 : 0;
 
-  my $is_target = ($metadata->is_ccs eq 'true' || @run_records > 1 ||
-             $is_r_and_d || $is_aux) ? 0 : 1;
+  # is_target = 0 logic - relies on all barcoded samples being 
+  # deplexed even if they are in a single sample pool. 
+  my $is_target = ((@run_records == 1 && $run_records[0]->tag_sequence) ||
+    ($type ne $CCS_SEQUENCE_PRODUCT && $metadata->is_ccs eq 'true') ||
+     @run_records > 1 || $is_r_and_d || $is_aux) ? 0 : 1;
```

Comment:
I think this could do with some exposition for future maintainers. I can see what tests are carried here, but would like some notes on why.

Reply:
Done. This is fragile and a better solution is on the todo list when we work towards not archiving non target BAM files in future.  Thankfully the is_target flag is not depended on downstream as far as I'm aware.